mlir/lib/Dialect/Utils/StructuredOpsUtils.cpp

```diff
@@ -120,6 +120,31 @@ bool mlir::isVecmat(ArrayAttr indexingMaps) {
   return indexingMaps == maps;
 }
 
+bool mlir::isBatchVecmat(ArrayAttr indexingMaps) {
+  if (indexingMaps.size() != 3)
+    return false;
+  AffineMap map0 = cast<AffineMapAttr>(indexingMaps[0]).getValue();
```

Comment:
you could use the `infer` utils which I find nicer to work with: https://github.com/llvm/llvm-project/blob/d1556e5efbf0cb671c0f6e403fc1eaf9153f8713/mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp#L1325 your call :)

Reply:
i see. nicer indeed, but i was copy and pasting other instances in this file. The move to `infer` looks like it would be a nice improvement, but should be done for the whole file togeteher, so, in a separate PR.